src/util.h

```diff
@@ -27,37 +27,28 @@
 #include <string>
 #include <vector>
 
-#include <boost/signals2/signal.hpp>
 #include <boost/thread/exceptions.hpp>
 
-/** Signals for translation. */
-class CTranslationInterface
-{
-public:
-    /** Translate a message to the native language of the user. */
-    boost::signals2::signal<std::string (const char* psz)> Translate;
-};
-
 extern std::map<std::string, std::string> mapArgs;
 extern std::map<std::string, std::vector<std::string> > mapMultiArgs;
 extern bool fDebug;
 extern bool fServer;
 
-extern CTranslationInterface translationInterface;
-
 [[noreturn]] extern void new_handler_terminate();
 
 extern const char * const BITCOIN_CONF_FILENAME;
 extern const char * const BITCOIN_PID_FILENAME;
 
+/** Translate a message to the native language of the user. */
+const extern std::function<std::string(const char*)> G_TRANSLATION_FUN;
+
 /**
- * Translation function: Call Translate signal on UI interface, which returns a boost::optional result.
- * If no translation slot is registered, nothing is returned, and simply return the input.
+ * Translation function.
+ * If no translation function is set, simply return the input.
  */
 inline std::string _(const char* psz)
 {
-    boost::optional<std::string> rv = translationInterface.Translate(psz);
-    return rv ? (*rv) : psz;
+    return G_TRANSLATION_FUN ? (G_TRANSLATION_FUN)(psz) : psz;
```

Comment:
Hm, now that I see how this is used I get it. I don't feel good about it though; it's more globals (ab)use.

Reply:
It's replacing one global with another, so it's not a regression in any case. But yes, this should be handled in a separate PR that updates the translation system (for example, my proposal to switch from `gettext` to `fluent`).